File: src/main/java/net/dontdrinkandroot/gitki/wicket/model/AbstractPathNameModel.java
package net.dontdrinkandroot.gitki.wicket.model;

import net.dontdrinkandroot.gitki.model.AbstractPath;
import net.dontdrinkandroot.wicket.model.AbstractChainedModel;
import org.apache.wicket.model.IModel;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class AbstractPathNameModel extends AbstractChainedModel<AbstractPath, String>
{
    public AbstractPathNameModel(IModel<? extends AbstractPath> parent)
    {
        super(parent);
    }

    @Override
    public String getObject()
    {
        AbstractPath path = this.getParentObject();
        if (path.isRoot()) {
            return "/";
        }

        return path.getName();
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/model/AbstractPathNameModel.kt
package net.dontdrinkandroot.gitki.wicket.model

import net.dontdrinkandroot.gitki.model.GitkiPath
import net.dontdrinkandroot.wicket.kmodel.KModel

class AbstractPathNameModel(private val parentModel: KModel<out GitkiPath>) : KModel<String> {

    override val value: String
        get() = parentModel.value.let { return if (it.root) "/" else it.name!! }

    override fun detach() {
        parentModel.detach()
    }
}